functorch/setup.py

```diff
@@ -1,149 +1,4 @@
-                for filename in glob.glob(wildcard):
-                    try:
-                        os.remove(filename)
-                    except OSError:
-                        shutil.rmtree(filename, ignore_errors=True)
-
-        # It's an old-style class in Python 2.7...
-        distutils.command.clean.clean.run(self)
-
-
-def get_extensions():
-    extension = CppExtension
-
-    # See functorch/csrc/Macros.h
-    define_macros = [('FUNCTORCH_BUILD_MAIN_LIB', None)]
-
-    extra_link_args = []
-    extra_compile_args = {"cxx": [
-        "-O3",
-        "-std=c++14",
-        "-fdiagnostics-color=always",
-    ]}
-    debug_mode = os.getenv('DEBUG', '0') == '1'
-    if debug_mode:
-        print("Compiling in debug mode")
-        extra_compile_args = {
-            "cxx": [
-                "-O0",
-                "-fno-inline",
-                "-g",
-                "-std=c++14",
-                "-fdiagnostics-color=always",
-            ]}
-        extra_link_args = ["-O0", "-g"]
-
-    this_dir = os.path.dirname(os.path.abspath(__file__))
-    extensions_dir = os.path.join(this_dir, "functorch", "csrc")
-
-    extension_sources = set(
-        os.path.join(extensions_dir, p)
-        for p in glob.glob(os.path.join(extensions_dir, "*.cpp"))
-    )
-    sources = list(extension_sources)
-    sources.append(os.path.join(extensions_dir, "dim", "dim.cpp"))
-
-    ext_modules = [
-        extension(
-            "functorch._C",
-            sources,
-            include_dirs=[this_dir],
-            define_macros=define_macros,
-            extra_compile_args=extra_compile_args,
-            extra_link_args=extra_link_args,
-        )
-    ]
-
-    return ext_modules
-
-
-class BuildExtension_(BuildExtension):
-    def build_extensions(self, *args, **kwargs):
-        # It turns out for windows this isn't populated?
-        if hasattr(self.compiler, 'compiler_so'):
-            if '-Wstrict-prototypes' in self.compiler.compiler_so:
-                self.compiler.compiler_so.remove('-Wstrict-prototypes')
-        super().build_extensions(*args, **kwargs)
-
-
-if __name__ == '__main__':
-    print("Building wheel {}-{}".format(package_name, version))
-    write_version_file()
-    setup(
-        # Metadata
-        name=package_name,
-        version=version,
-        author='PyTorch Core Team',
-        url="https://github.com/pytorch/functorch",
-        description='JAX-like composable function transforms for PyTorch',
-        license='BSD',
-
-        # Package info
-        packages=find_packages(),
-        install_requires=requirements,
-        extras_require=extras,
-        ext_modules=get_extensions(),
-        cmdclass={
-            "build_ext": BuildExtension_.with_options(no_python_abi_suffix=True),
-            'clean': clean,
-        })
+# Welcome to the FuncTorch setup.py.
+# As it has been embedded into PyTorch it's not supposed to do anything
+if __name__ == "__main__":
+    print("Mostly harmless")
```

Comment:
@malfet Do you think we could (or should) change this to https://github.com/pytorch/functorch/blob/e108ed74228ef4db8200b1df59138b1696ad573e/setup.py ?

The context is, there are at least three projects that install functorch via the following command: https://github.com/pytorch/torchdynamo/blob/d88a0ac0e4a799617e59a05fec79d329781f3ea7/Makefile#L54-L55 . It would be best if we don't hard-break them and I can migrate them off one-by-one after we do the merge

Reply:
Sure, let me update it like that.